src/modules/workspaces/workspaces.usecase.spec.ts

```diff
@@ -949,4 +950,441 @@ describe('WorkspacesUsecases', () => {
+        const teamId = 'team-id';
+        const memberId = 'member-uuid';
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(null);
+
+        await expect(
+          service.removeMemberFromTeam(teamId, memberId),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is not part of team, then it should throw', async () => {
+        const user = newUser();
+        const workspace = newWorkspace();
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+        });
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(null);
+
+        await expect(
+          service.removeMemberFromTeam(team.id, user.uuid),
+        ).rejects.toThrow(BadRequestException);
+      });
+
+      it('When user is the manager of the team, then owner of the workspace should be assigned as manager', async () => {
+        const workspaceOwner = newUser();
+        const workspace = newWorkspace({ owner: workspaceOwner });
+        const manager = newUser();
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+          manager: manager,
+        });
+        const teamUser = newWorkspaceTeamUser({
+          teamId: team.id,
+          memberId: manager.uuid,
+        });
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(teamUser);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+
+        await service.removeMemberFromTeam(team.id, manager.uuid);
+
+        expect(teamRepository.updateById).toHaveBeenCalledWith(team.id, {
+          managerId: workspace.ownerId,
+        });
+      });
+
+      it('When user is being removed, then it should resolve', async () => {
+        const workspaceOwner = newUser();
+        const workspace = newWorkspace({ owner: workspaceOwner });
+        const member = newUser();
+        const team = newWorkspaceTeam({
+          workspaceId: workspace.id,
+        });
+        const teamUser = newWorkspaceTeamUser({
+          teamId: team.id,
+        });
+
+        jest.spyOn(teamRepository, 'getTeamById').mockResolvedValue(team);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+        jest.spyOn(teamRepository, 'getTeamUser').mockResolvedValue(teamUser);
+        jest.spyOn(workspaceRepository, 'findOne').mockResolvedValue(workspace);
+
+        await service.removeMemberFromTeam(team.id, member.uuid);
+
+        expect(teamRepository.removeMemberFromTeam).toHaveBeenCalledWith(
+          team.id,
+          member.uuid,
+        );
+      });
+    });
+
+    describe('changeTeamManager', () => {
```

Comment:
What happens if the team manager that is going to be changed is the manager of the default team? (In this case, the owner?

Reply:
It won't proceed if deault team = true